bridges/GQMagazineBridge.php

```diff
@@ -62,30 +78,31 @@ public function collectData()
 		$main = $html->find('main', 0);
 		foreach ($main->find('a') as $link) {
 			$uri = $link->href;
-			$title = $link->find('h2', 0);
 			$date = $link->find('time', 0);
 
 			$item = array();
 			$author = $link->find('span[itemprop=name]', 0);
-			$item['author'] = $author->plaintext;
-			$item['title'] = $title->plaintext;
-			if(substr($uri, 0, 1) === 'h') { // absolute uri
-				$item['uri'] = $uri;
-			} else if(substr($uri, 0, 1) === '/') { // domain relative url
-				$item['uri'] = $this->getDomain() . $uri;
-			} else {
-				$item['uri'] = $this->getDomain() . '/' . $uri;
-			}
+			if($author != null) {
+				$item['author'] = $author->plaintext;
+				$item['title'] = $this->findTitleOf($link);
+				if(substr($uri, 0, 1) === 'h') { // absolute uri
```

Comment:
This could be a good spot for a switch statement

```
switch(substr($uri, 0, 1)) {
	case 'h':
		$item['uri'] = $uri;
		break;
	case '/':
		$item['uri'] = $this->getDomain() . $uri;
		break;
	default:
		$item['uri'] = $this->getDomain() . '/' . $uri;
}
```

Reply:
good idea !